lua/tests/cfc_restrict_materials/main.lua

```diff
@@ -0,0 +1,25 @@
+local function checkMat( mat )
+    local prop = ents.Create( "prop_physics" )
+    local mat1 = prop:GetMaterial()
+    prop:SetMaterial( mat )
+    local mat2 = prop:GetMaterial()
+
+    prop:Remove()
+    return mat1 == mat2
+end
+return {
+    {
+        name = "Check if allowed materials are properly changed.",
+        func = function()
+            local same = checkMat( "brick/brick_model" )
+            expect( !same )
+        end
+    },
+    {
+        name = "Check if not materials are not applied.",
+        func = function()
+            local same = checkMat( "pp/copy" )
+            expect( same )

```

Comment:
```suggestion
            local mat = "pp/copy"
            expect( checkMat( mat ) ).to.beFalse()
```

Reply:
horror

Reply:
Normally you'd do the entity creation and checking and such inside of the test, but because we don't have a mock library or anything I'll let the helper function slide. This is the result of that.

It could've been
```lua
expect( ent:GetMaterial() ).to.eq( testMaterial )
```